File: json-supabase/nodejs-contacts-uploader/upload_contacts.js
#!/usr/bin/env node
/**
 * Script to upload contacts from contacts.json to Supabase
 */

const fs = require('fs');
const { createClient } = require('@supabase/supabase-js');
require('dotenv').config();

// Load contacts from JSON file
function loadContacts() {
    try {
        const data = fs.readFileSync('contacts.json', 'utf8');
        return JSON.parse(data);
    } catch (error) {
        if (error.code === 'ENOENT') {
            console.error('Error: contacts.json file not found!');
        } else if (error instanceof SyntaxError) {
            console.error('Error: Invalid JSON in contacts.json file!');
        } else {
            console.error('Error reading file:', error.message);
        }
        return null;
    }
}

// Upload contacts to Supabase
async function uploadToSupabase(contacts) {
    // Get Supabase credentials from environment variables
    const supabaseUrl = process.env.SUPABASE_URL;
    const supabaseKey = process.env.SUPABASE_ANON_KEY;
    
    if (!supabaseUrl || !supabaseKey) {
        console.error('Error: Missing Supabase credentials!');
        console.error('Please set SUPABASE_URL and SUPABASE_ANON_KEY environment variables');
        return false;
    }
    
    try {
        // Initialize Supabase client
        const supabase = createClient(supabaseUrl, supabaseKey);
        
        // Upload contacts to the 'contacts' table
        const { data, error } = await supabase
            .from('contacts')
            .insert(contacts);
        
        if (error) {
            console.error('Supabase error:', error);
            return false;
        }
        
        console.log(`Successfully uploaded ${contacts.length} contacts to Supabase!`);
        console.log('Response:', data);
        return true;
        
    } catch (error) {
        console.error('Error uploading to Supabase:', error.message);
        return false;
    }
}

// Main function
async function main() {
    console.log('Loading contacts from contacts.json...');
    const contacts = loadContacts();
    
    if (!contacts) {
        return;
    }
    
    console.log(`Found ${contacts.length} contacts`);
    
    // Display first contact as preview
    if (contacts.length > 0) {
        console.log('\nFirst contact preview:');
        console.log(JSON.stringify(contacts[0], null, 2));
    }
    
    // Ask for confirmation
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });
    
    rl.question('\nDo you want to upload these contacts to Supabase? (y/n): ', async (answer) => {
        rl.close();
        
        if (answer.toLowerCase() !== 'y') {
            console.log('Upload cancelled.');
            return;
        }
        
        // Upload to Supabase
        console.log('\nUploading to Supabase...');
        const success = await uploadToSupabase(contacts);
        
        if (success) {
            console.log('✅ Upload completed successfully!');
        } else {
            console.log('❌ Upload failed!');
        }
    });
}

// Run the script
if (require.main === module) {
    main().catch(console.error);
}

File: json-supabase/python-contacts-uploader/upload_contacts.py
#!/usr/bin/env python3
"""
Script to upload contacts from contacts.json to Supabase
"""

import json
import os
from supabase import create_client, Client
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

def load_contacts():
    """Load contacts from the JSON file"""
    try:
        with open('contacts.json', 'r') as file:
            return json.load(file)
    except FileNotFoundError:
        print("Error: contacts.json file not found!")
        return None
    except json.JSONDecodeError:
        print("Error: Invalid JSON in contacts.json file!")
        return None

def upload_to_supabase(contacts):
    """Upload contacts to Supabase"""
    # Get Supabase credentials from environment variables
    supabase_url = os.getenv('SUPABASE_URL')
    supabase_key = os.getenv('SUPABASE_ANON_KEY')
    
    if not supabase_url or not supabase_key:
        print("Error: Missing Supabase credentials!")
        print("Please set SUPABASE_URL and SUPABASE_ANON_KEY environment variables")
        return False
    
    try:
        # Initialize Supabase client
        supabase: Client = create_client(supabase_url, supabase_key)
        
        # Upload contacts to the 'contacts' table
        result = supabase.table('contacts').insert(contacts).execute()
        
        print(f"Successfully uploaded {len(contacts)} contacts to Supabase!")
        print(f"Response: {result}")
        return True
        
    except Exception as e:
        print(f"Error uploading to Supabase: {e}")
        return False

def main():
    """Main function"""
    print("Loading contacts from contacts.json...")
    contacts = load_contacts()
    
    if contacts is None:
        return
    
    print(f"Found {len(contacts)} contacts")
    
    # Display first contact as preview
    if contacts:
        print("\nFirst contact preview:")
        print(json.dumps(contacts[0], indent=2))
    
    # Ask for confirmation
    response = input("\nDo you want to upload these contacts to Supabase? (y/n): ")
    if response.lower() != 'y':
        print("Upload cancelled.")
        return
    
    # Upload to Supabase
    print("\nUploading to Supabase...")
    success = upload_to_supabase(contacts)
    
    if success:
        print("✅ Upload completed successfully!")
    else:
        print("❌ Upload failed!")

if __name__ == "__main__":
    main()
